classes/admin_settings/multi_checkbox_with_icon.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace block_sharing_cart\admin_settings;
+
+// @codeCoverageIgnoreStart
+defined('MOODLE_INTERNAL') || die();
+
+// @codeCoverageIgnoreEnd
+
+class multi_checkbox_with_icon extends \admin_setting_configmulticheckbox
+{
+    protected array $icons;
+
+    public function __construct(
+        string $name,
+        string $visiblename,
+        string $description,
+        $defaultsetting,
+        array $choices,
+        array $icons
+    ) {
+        $this->icons = $icons;
+        parent::__construct($name, $visiblename, $description, $defaultsetting, $choices);
+    }
+
+    public function output_html($data, $query = ''): string
+    {
+        if (empty($this->choices) || !$this->load_choices()) {
+            return '';
+        }
+        $default = $this->get_defaultsetting();
+        if (is_null($default)) {
+            $default = [];
+        }
+        if (is_null($data)) {
+            $data = [];
+        }
+        $options = [];
+        $defaults = [];
+        foreach ($this->choices as $key => $description) {
+            if (!empty($data[$key])) {
+                $checked = 'checked="checked"';
+            } else {
+                $checked = '';
+            }
+            if (!empty($default[$key])) {
+                $defaults[] = $description;
+            }
+
+            $options[] = '<input type="checkbox" id="' . $this->get_id(
+                ) . '_' . $key . '" name="' . $this->get_full_name(
+                ) . '[' . $key . ']" value="1" ' . $checked . ' class="mr-1"/>' . '<label for="' . $this->get_id(
+                ) . '_' . $key . '">' . $this->icons[$key] . highlightfast($query, $description) . '</label>';
+        }
+
+        if (is_null($default)) {
+            $defaultinfo = null;
+        } else {
+            if (!empty($defaults)) {
+                $defaultinfo = implode(', ', $defaults);
+            } else {
+                $defaultinfo = get_string('none');
+            }
+        }
+
+        $return = '<div class="form-multicheckbox">';
+        $return .= '<input type="hidden" name="' . $this->get_full_name(
+            ) . '[xxxxx]" value="1" />'; // something must be submitted even if nothing selected
+        if ($options) {
+            $return .= '<ul>';
+            foreach ($options as $option) {
+                $return .= '<li>' . $option . '</li>';
+            }
+            $return .= '</ul>';
+        }
+        $return .= '</div>';
```

Comment:
Wouldn't it be possible to add this to a mustache template?
And what is the [xxxxx]? Missing dynamic key?

```mustache
<div class="form-multicheckbox">
  <input type="hidden" name"{{full_name}}[xxxxx]" value="1" />
  {{#has_options}}
    <ul>
      {{#options}}
        <li>
          <input type="checkbox" id="{{option_id}}_{{option_key}}" name="{{option_name}}_[{{option_key}}]"
          value="1" {{option_checked}} class="mr-1"/>
          <label for="{{option_id}}_{{option_key}}">{{label_name}}</label>
        </li>
      {{/options}}
    </ul>
  {{/has_options}}
</div>
```

Reply:
> Wouldn't it be possible to add this to a mustache template? 

I'm unsure whether this will make it into the sharing cart. It's part of the old settings that we may just remove entirely.

> And what is the [xxxxx]? Missing dynamic key?

I have no clue